### PR TITLE
fix: move util functions with single ref

### PR DESCRIPTION
**Pull request by peternhale**
@W-11748691@

### CODE REVIEW

File: src/package/packageVersionReport.ts [hunk 1]
```diff
@@ -32,6 +32,51 @@ const QUERY_VERBOSE =
...
+
+/**
+ * Given a list of subscriber package version IDs (04t), return the associated version strings (e.g., Major.Minor.Patch.Build)
+ *
+ * @return Map of subscriberPackageVersionId to versionString
+ * @param subscriberPackageVersionIds
+ * @param connection For tooling query
+ */
+// eslint-disable-next-line @typescript-eslint/require-await
+async function getPackageVersionStrings(
```
Comment (WillieRuemmele) [inline src/package/packageVersionReport.ts:44]: why keep this async, if it's not required to be?

**Review by WillieRuemmele — APPROVED** [review]

File: src/package/packageVersionReport.ts (hunk 1 above)
Comment (peternhale) [reply src/package/packageVersionReport.ts:44]: I believe I need to remove the eslint directive. Line 63 is using await.

File: test/package/packageVersionCreate.test.ts [hunk 2]
```diff
@@ -61,7 +61,7 @@ describe('Package Version Create', () => {
     $$.SANDBOX.stub(xml2js, 'parseStringPromise').resolves({
       Package: { types: [{ name: ['Apexclass'], members: ['MyApexClass'] }] },
     });
-    $$.SANDBOX.stub(pkgUtils, 'getPackageType').resolves('Managed');
+    packageTypeStud = $$.SANDBOX.stub(pkgUtils, 'getPackageType').resolves('Managed');
```
Comment (shetzel) [inline test/package/packageVersionCreate.test.ts:64]: nit: `packageTypeStub`

File: test/package/packageVersionCreate.test.ts [hunk 3]
```diff
@@ -263,9 +263,10 @@ describe('Package Version Create', () => {
   it('should validate options when package type = unlocked (scripts)', async () => {
     packageTypeQuery.restore();
     // @ts-ignore
-    packageTypeQuery = $$.SANDBOX.stub(connection.tooling, 'query').resolves({
-      records: [{ ContainerOptions: 'Unlocked' }],
-    });
+    packageTypeQuery = $$.SANDBOX.stub(connection.tooling, 'query')
```
Comment (shetzel) [inline test/package/packageVersionCreate.test.ts:266]: I _think_ we can remove the ts-ignores if we use `stubMethod` from ts-sinon.  Perhaps another effort as well...

File: src/package/packageInstall.ts [hunk 4]
```diff
@@ -192,6 +216,20 @@ async function pollStatus(
...
+export function isErrorFromSPVQueryRestriction(err: Error): boolean {
+  return (
+    err.name === 'MALFORMED_QUERY' &&
+    err.message.includes('Implementation restriction: You can only perform queries of the form Id')
+  );
+}
+
+export function isErrorPackageNotAvailable(err: Error): boolean {
+  return err.name === 'UNKNOWN_EXCEPTION' || err.name === 'PACKAGE_UNAVAILABLE';
+}
```
[the remark this answers is not in the document]
Comment (peternhale) [reply src/package/packageInstall.ts:231]: No, will fix

File: src/package/packageVersionCreate.ts [hunk 5]
```diff
@@ -682,4 +693,283 @@ export class PackageVersionCreate {
...
+    // remove references to Second Generation
+    if (err.message.includes('Second Generation ')) {
+      err['message'] = err.message.replace('Second Generation ', '');
+    }
+
+    return err;
+  }
+
+  // eslint-disable-next-line complexity
+  private async getAncestorId(
```
[the remark this answers is not in the document]
Comment (peternhale) [reply src/package/packageVersionCreate.ts:765]: Ancestry is complicated, so may be worth a look.

File: src/package/packageVersionReport.ts [hunk 6]
```diff
@@ -32,6 +32,50 @@ const QUERY_VERBOSE =
...
+ * @param connection For tooling query
+ */
+async function getPackageVersionStrings(
+  subscriberPackageVersionIds: string[],
+  connection: Connection
+): Promise<Map<string, string>> {
+  type PackageVersionString = Pick<
+    PackagingSObjects.Package2Version,
+    'SubscriberPackageVersionId' | 'MajorVersion' | 'MinorVersion' | 'PatchVersion' | 'BuildNumber'
+  >;
```
[the remark this answers is not in the document]
Comment (peternhale) [reply src/package/packageVersionReport.ts:50]: I agree that we should expose useful types, but am conflicted. This type unique to this file and all the places it is used are internal. So we need to keep this in mind when deciding on what to export.

File: test/package/packageVersionCreate.test.ts (hunk 3 above)
Comment (peternhale) [reply test/package/packageVersionCreate.test.ts:266]: +1 on next round.

File: src/package/packageConvert.ts [hunk 7]
```diff
@@ -31,14 +31,49 @@ import {
...
-import { convertCamelCaseStringToSentence, generatePackageAliasEntry } from '../utils';
+import { generatePackageAliasEntry } from '../utils';
 import { byId } from './packageVersionCreateRequest';
 import * as pvcr from './packageVersionCreateRequest';
 import Package2VersionStatus = PackagingSObjects.Package2VersionStatus;
 
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/packaging', 'package_version_create');
 
+async function findOrCreatePackage2(seedPackage: string, connection: Connection): Promise<string> {
```
[the remark this answers is not in the document]
Comment (peternhale) [reply src/package/packageConvert.ts:42]: This is context specific, "see if the package I want to convert has already been converted (only allowed to convert a specific package once), if so return the converted package id, other wise get the get the package by subscriber Id and then create a new package." I can see the guts changing to Package.find(...), Subscriber.find() and then package.create()"

[the remark this answers is not in the document]
Comment (shetzel) [reply src/package/packageConvert.ts:42]: ahh ok, that makes sense

File: src/package/packageVersionReport.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (shetzel) [reply src/package/packageVersionReport.ts:50]: Ok if that's the case (not useful externally) then keep it.

File: test/package/packageConvert.test.ts [hunk 8]
```diff
@@ -8,8 +8,12 @@ import { expect } from 'chai';
...
 import { Connection, Lifecycle } from '@salesforce/core';
 import { Duration } from '@salesforce/kit';
-import { convertPackage, createPackageVersionCreateRequest } from '../../src/package/packageConvert';
-import * as pkgUtils from '../../src/utils';
+import {
+  convertPackage,
+  createPackageVersionCreateRequest,
+  findOrCreatePackage2,
+} from '../../src/package/packageConvert';
+// import * as pkgUtils from '../../src/utils';
```
Comment (shetzel) [inline test/package/packageConvert.test.ts:16]: This commented import can be removed now correct?

**Review by shetzel — APPROVED** [review]

**Pull request merged**